upstream-sha256: 813a8753fd1de44a10117ab3df6be2e87806dae9c2b7e9f543b307eb4f59d3e4
actual-sha256: e664eb45129976b1b63913c9cb989eef209a9daeaad2f7aec573baf7355dbfbb
--- upstream.expected
+++ tolk.actual
@@ -14,7 +14,7 @@
   float val5 = (*(data1_512+(alu0+384)));
   float val6 = (*(data1_512+(alu0+448)));
   float val7 = (*(data1_512+(alu0+64)));
-  *(data0_64+alu0) = (val0+val7+val1+val2+val3+val4+val5+val6);
+  *(data0_64+alu0) = (0.0f+val0+val7+val1+val2+val3+val4+val5+val6);
 }
 ---
 typedef unsigned int uint;
@@ -32,17 +32,6 @@
 #define INFINITY (__int_as_float(0x7f800000))
 #define NAN (__int_as_float(0x7fffffff))
 template <class T, class F> __device__ __forceinline__ T tg_bitcast(F v) { union U { F f; T t; }; U u; u.f = v; return u.t; }
-extern "C" __global__ void __launch_bounds__(8) E_8_4(float* data0_32, float* data1_32) {
-  int lidx0 = threadIdx.x; /* 8 */
-  int alu0 = (lidx0<<2);
-  float4 val0 = (*((float4*)((data1_32+alu0))));
-  *((float4*)((data0_32+alu0))) = val0;
-}
----
-typedef unsigned int uint;
-#define INFINITY (__int_as_float(0x7f800000))
-#define NAN (__int_as_float(0x7fffffff))
-template <class T, class F> __device__ __forceinline__ T tg_bitcast(F v) { union U { F f; T t; }; U u; u.f = v; return u.t; }
 extern "C" __global__ void __launch_bounds__(8) E_8_4(float* data0_32, float* data1_32, float* data2_64) {
   int lidx0 = threadIdx.x; /* 8 */
   int alu0 = (lidx0<<2);
@@ -150,17 +139,6 @@
   int lidx0 = threadIdx.x; /* 8 */
   int alu0 = (lidx0<<2);
   float4 val0 = (*((float4*)((data1_64+(alu0+32)))));
-  *((float4*)((data0_32+alu0))) = val0;
-}
----
-typedef unsigned int uint;
-#define INFINITY (__int_as_float(0x7f800000))
-#define NAN (__int_as_float(0x7fffffff))
-template <class T, class F> __device__ __forceinline__ T tg_bitcast(F v) { union U { F f; T t; }; U u; u.f = v; return u.t; }
-extern "C" __global__ void __launch_bounds__(8) E_8_4(float* data0_32, float* data1_32) {
-  int lidx0 = threadIdx.x; /* 8 */
-  int alu0 = (lidx0<<2);
-  float4 val0 = (*((float4*)((data1_32+alu0))));
   *((float4*)((data0_32+alu0))) = val0;
 }
 ---
